resources/sts/4.12/hypershift/InstallerPolicy.json

```diff
@@ -4,67 +4,130 @@
-                "iam:CreatePolicy",
-                "iam:CreatePolicyVersion",
-                "iam:CreateRole",
-                "iam:DeleteOpenIDConnectProvider",
-                "iam:DeletePolicyVersion",
-                "iam:DeleteRole",
-                "iam:DeleteRolePermissionsBoundary",
-                "iam:DetachRolePolicy",
-                "iam:GetOpenIDConnectProvider",
-                "iam:GetPolicy",
-                "iam:GetRole",
-                "iam:GetUser",
-                "iam:ListAttachedRolePolicies",
-                "iam:ListPolicyTags",
-                "iam:ListPolicyVersions",
-                "iam:ListRoleTags",
-                "iam:PutRolePermissionsBoundary",
-                "iam:TagPolicy",
                 "kms:DescribeKey",
-                "route53:ChangeResourceRecordSets",
-                "route53:DeleteHostedZone",
-                "route53:GetHostedZone",
+                "iam:GetRole",
                 "route53:ListHostedZones",
-                "route53:ListHostedZonesByName",
                 "route53:ListResourceRecordSets",
-                "s3:CreateBucket",
-                "s3:DeleteBucket",
-                "s3:DeleteObject",
-                "s3:ListObjects",
-                "s3:PutObject",
-                "secretsmanager:GetSecretValue",
                 "servicequotas:GetServiceQuota",
-                "sts:AssumeRole",
-                "sts:AssumeRoleWithWebIdentity",
                 "sts:GetCallerIdentity"
             ],
-        "Resource": "*"
+            "Resource": "*"
+        },
+        {
+            "Action": [
+                "iam:PassRole"
+            ],
+            "Resource": [
+                "arn:*:iam::*:role/*-Worker-Role"
+            ],
+            "Effect": "Allow"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+              "sts:AssumeRole",
+              "sts:AssumeRoleWithWebIdentity"
+            ],
+            "Resource": [
+              "arn:aws:iam::*:role/*"
+            ],
+            "Condition": {
+              "StringEquals": {
+                "aws:ResourceTag/red-hat-managed": "true"
+              }
+            }
+        },
+        {
+            "Effect": "Allow",
+            "Action": "elasticloadbalancing:CreateLoadBalancer",
+            "Resource": [
+              "*"
+            ],
+            "Condition": {
+              "StringEquals": {
+                "aws:RequestTag/red-hat-managed": "true"
+              }
+            }
+        },
+        {
+            "Sid": "BucketActions",
+            "Effect": "Allow",
+            "Action": [
+                "s3:CreateBucket",
+                "s3:DeleteBucket"
+            ],
+            "Resource": [
+                "arn:aws:s3:::*-oidc"
+            ]
+        },
+        {
+            "Sid": "CreateAndManageInstanceProfiles",
+            "Effect": "Allow",
+            "Action": [
+                "iam:AddRoleToInstanceProfile",
```

Comment:
We should be able to restrict these by tag, did you try it? If not we'll need a card for the HyperShift team

Reply:
I think we can restrict the `CreateInstanceProfile` action using condition keys for aws tags, however other actions do not seem to support condition keys as per [docs](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsidentityandaccessmanagement.html) ?

In any case, the instance profiles are not being tagged with `red-hat-managed=true` at the moment :
```
$ aws iam list-instance-profile-tags --instance-profile-name staging-ID-cluster-worker
{
    "Tags": [],
    "IsTruncated": false
}
```

I have created https://issues.redhat.com/browse/SDA-8142 to get this fixed and a follow up for us to edit this policy once the tags are set (OSD-15178) so that it doesnt block this PR.

Thanks, this is a good spot.